services:
  db:
    image: postgres:17
    container_name: postgres_db
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - app_network
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
      - data:/var/lib/postgresql/data


  backend:
    image: app_backend 
    build:
      context: ./cutaway
    container_name: app_backend
    restart: always
    depends_on:
      - db
    environment:
      DATABASE_URL: postgresql+asyncpg://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db:5432/${POSTGRES_DB}
      ALEMBIC_DB_URL: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db:5432/${POSTGRES_DB}
    ports:
      - "8080:8080"
    volumes:
      - .:/cutaway
      - ./cutaway/static/img/cards:/app/static/img/cards
    command: ["python", "main.py"]
    networks:
      - app_network

volumes:
  data:
  images:

networks:
  app_network:
    driver: bridge
